ethportal-api/src/types/network.rs

```diff
@@ -29,32 +33,109 @@ impl std::str::FromStr for Network {
 }
 
 /// Enum for various different portal subnetworks in a "core" network.
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
 pub enum Subnetwork {
     Beacon,
     History,
     State,
+    CanonicalIndices,
+    VerkleState,
+    TransactionGossip,
+    Utp,
 }
 
 impl fmt::Display for Subnetwork {
```

Comment:
It's a little awkward here for these str conversion traits...

For `Protocolid`, we used `Display` & `FromStr` to convert cli args (aka camel case), so I implemented some helper functions to be used in these scenarios.

Ok, we could use the `Debug` trait as the "pretty" version, (the only difference being the extra space in multi-letter variants). And then use `Display`/`FromStr` to handle cli args. But imo `Display` is meant to be the "pretty" version of the string.

But, we also use `Display` to create the "metrics" label. And, if we change that, then there's a whole lot of following changes that need to be made to our dashboard

Reply:
I like current approach.

I just wanted to mention that we also have [alternate formatting](https://doc.rust-lang.org/std/fmt/#sign0) at our disposal. I don't think it really fits for this use case, but you see a nice way to use it.